tests/autoyast/verify_ext4.pm

```diff
@@ -0,0 +1,61 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate autoyast_ext4 profile.
+#          * Verify registration
+#          * Verify partitioning: ext4 and swap
+#          * Verify software packages/patterns
+#          * Verify users
+# Maintainer: Joaquín Rivera <jeriveramoya@suse.com>
+
+use base 'basetest';
+use strict;
+use testapi;
+
+sub verify_registration {
+    my $output = script_output 'SUSEConnect --list-extensions';
+    unless ($output =~ /Basesystem.*Activated/ && $output =~ /Applications Server.*Activated/) {
+        die 'Registered system does not contains Basesystem and/or Applications Server';
+    }
+}
+
+sub verify_partitioning {
+    assert_script_run 'blkid /dev/vda | grep PTTYPE="dos"';
+    assert_script_run 'grep "swap\s\+swap" /etc/fstab';
+    assert_script_run 'grep "/\s\+ext4" /etc/fstab';
+}
+
+sub verify_software {
+    my @packages = qw(glibc grub2 snapper syslinux kdump sles-release kexec-tools
```

Comment:
I would not do such check, as we have constantly changing list of packages and patterns get renamed, so it will be a lot of efforts to maintain. If you feel fancy, we can have list of packages as variable which is used both in the profile and in this test. Also, keep in mind that we have different list for SLE 12 and SLE 15.